python/tank/authentication/shotgun_wrapper.py

```diff
@@ -60,6 +63,21 @@ def _call_rpc(self, *args, **kwargs):
         except AuthenticationFault:
             logger.debug("Authentication failure.")
             pass
+        except ProtocolError as e:
```

Comment:
The way things were being handled in this file, I figured that this is where I should attempt the renew_session.  It felt bad throwing an exception to deal with that at a different point.